### PR TITLE
dev: chg: solve vulns

### PR DESCRIPTION
**Pull request by marcello33**
This PR bumps some dependencies to solve reported vulnerabilities, and change code in some places to make it compatible with new libraries.

### CODE REVIEW

**Review by Raneet10 — DISMISSED** [review]

File: crypto/keys/hd/fundraiser_test.go [hunk 1]
```diff
@@ -72,7 +72,6 @@ func TestFundraiserCompatibility(t *testing.T) {
 		require.Equal(t, priv[:], privB, "Expected priv keys to match")
 		var pubBFixed [33]byte
 		copy(pubBFixed[:], pubB)
-		require.Equal(t, pub, secp256k1.PubKeySecp256k1(pubBFixed), fmt.Sprintf("Expected pub keys to match for %d", i))
```
Comment (ssandeep) [inline crypto/keys/hd/fundraiser_test.go:75]: Any alternate tests required after removing this line?

Comment (marcello33) [reply crypto/keys/hd/fundraiser_test.go:75]: Yeah but need to reimplement the PubKey spec according to the new library, which would take some time

Comment (marcello33) [reply crypto/keys/hd/fundraiser_test.go:75]: Also this hd `fundraiser_test` was removed from `cosmos-sdk` upstream

**Review by VAIBHAVJINDAL3012 — APPROVED** [review]

File: crypto/keys/hd/fundraiser_test.go (hunk 1 above)
Comment (marcello33) [reply crypto/keys/hd/fundraiser_test.go:75]: Just to be on the safe side @ssandeep 
I checked it against `sidechannel` branch (default branch of our fork) and this test fails there too.
The thing is that it was never adjusted to work with uncompressed (eth standard) key format.
[Here](https://github.com/maticnetwork/cosmos-sdk/blob/sidechannel/crypto/keys/hd/fundraiser_test.go#L73) it's still using 33 bytes but [underlying peppermint](https://github.com/maticnetwork/tendermint/blob/peppermint/crypto/secp256k1/secp256k1.go#L151) expects 65 bytes

**Pull request merged**